### DISCUSSION TITLE
Chapter08 -- wikipedia URL not working

### DISCUSSION THREAD

**Discussion by LucaGiona** [headfirstpython/third#9] [Show and tell]
Hi I had a problem with the URL -- I always got the message after running this Jupyter Notebook cell:
```
html = gazpacho.get(URL)
```
**URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1020)>**

After searching I found this solution that worked for me:
```
**import ssl**
import gazpacho


**ssl._create_default_https_context = ssl._create_unverified_context**

URL = "https://en.wikipedia.org/wiki/List_of_world_records_in_swimming"
html = gazpacho.get(URL)
len(html)
```
using the highlighted (in different cells) worked for me --(leave the ** out)

**Comment by headfirstpython** [comment]
I suspect this might be an issue with certificates not being installed properly/correctly in your operating system, as you should not need to use the `ssl` library in this way to get this code to execute.  On my Mac, there's a package called `ca-certificates` which (I think) does this for me.  With all of my usages of `gazpacho` over the years, together with the experiences of hundreds of my students (using mainly Windows but also some Linux and Mac), running this code has never been an issue. Thanks.

**Comment by LucaGiona** [comment]
Thank you for your response and clarification. I completely understand that this is likely an issue specific to my environment, as you and your students haven’t experienced it before. I truly didn’t mean to criticize the code or the book—it’s been incredibly helpful!

I’ve tried reinstalling the certificates on my system, but unfortunately, that didn’t resolve the issue. I suspect the problem may require a full reinstallation of Python 3 on my machine, so I’ll look into that next.

Just to clarify, I opened this discussion on GitHub to share my experience, in case someone else encounters a similar problem. I know the book itself is not the issue, and I wanted to provide a possible workaround for others who might face the same challenge.

Honestly, the book is excellent overall! However, I noticed there aren’t many forks on GitHub, which might suggest it isn’t widely viewed. Perhaps that’s another reason this issue hasn’t come up before.

Thank you again for your time and your response!

**Comment by headfirstpython** [comment]
Hey, no worries.  I very much appreciate you sharing your experience, and I am chuffed that you are enjoying my book.  And, of course, everyone's experience is different and weird things often trip-up all of us.  I am sorry to see that the certificates aren't the issue... and I'd be interested to learn if reinstalling Python makes a difference. As you said, sharing this sort of activity often helps other readers with similar problems.  Using GitHub in this way is very much a new thing with this edition.  Most readers (typically) register issues or problems with the book's content via O'Reilly's Errata system (which can have it's advantages and disadvantages). My hope is that more readers will seek me out on GitHub as time passes.

Thanks! 😎
